internal/archive/archive.go

```diff
@@ -184,7 +184,8 @@ func WriteDirToTar(tw TarWriter, srcDir, basePath string, uid, gid int, mode int
 				return err
 			}
 
-			header, err = tar.FileInfoHeader(fi, target)
+			// Ensure that symlinks have Linux link names, independent of source OS
+			header, err = tar.FileInfoHeader(fi, filepath.ToSlash(target))
```

Comment:
This doesn't seem quite right because we already transform the path at line 206. Could you explain why this exception is necessary just for symlinks here and not on line 193?

Reply:
It's necessary, because in getting the head path (the so-called "target"), we call `os.Readlink()`. That has differing implementations for `unix` and `windows` systems, and returns the OS dependent paths. 

With re: to line 206, when we change the path names in 206, we're doing it for `header.Name`, but not `header.Linkname` (only valid for symlinks), which is set using the `tar.FileInfoHeader` method here. This is a fix for a bug that came out when I checked the symlinks read from zip files, and saw that the Linknames was wrong.

Reply:
> It's necessary, because in getting the head path (the so-called "target"), we call os.Readlink(). That has differing implementations for unix and windows systems, and returns the OS dependent paths.

I don't know how relevant that is here since `fi` and `file` would also be OS dependent anywhere within this block.

> With re: to line 206, when we change the path names in 206, we're doing it for header.Name, but not header.Linkname (only valid for symlinks), which is set using the tar.FileInfoHeader method here. This is a fix for a bug that came out when I checked the symlinks read from zip files, and saw that the Linknames was wrong.

Ah, I see. It's all about `header.Linkname`. That's not very obvious. I wonder if it would be more obvious to add the following below line 206 instead:

```golang
if header.Linkname != "" {
  header.Linkname = filepath.ToSlash(header.Linkname)
}
```

Reply:
We could, but Linkname is only used in a symlink case, which is why I had just put it in the symlink Header initialization. Happy to move it there, though, if that would be a clearer place.

Reply:
I do think it would help but not a blocker now that I understand it's purpose.